### ISSUE TITLE
Grouping level for Rucio rules

### ISSUE THREAD

**Issue by drkovalskyi** [dmwm/WMCore#10952]
**Impact of the new feature**
WMAgent, MicroServices, T0

**Is your feature request related to a problem? Please describe.**
Operations would like to avoid spreading datasets across multiple sites unless they are very big. With data spread across multiple sites we increase risks of workflows getting stuck in a partially processed state when sites have issues. This leads to increased number of incomplete workflows in processing keeping disk space occupied for longer than necessary.

**Describe the solution you'd like**
In rucio.add_replication_rule calls set grouping = 'ALL' for all datasets smaller than 100TB and grouping = 'DATASET' otherwise. 

FYI @germanfgv, @haozturk

**Comment by haozturk** [comment]
Site selection algorithm would be quite important in case grouping = 'ALL'. Because if the input(s) of some high priority workflow(s) go to a site where computing capacity is low, we would see a higher turnaround time. If such a site has issues, this would get worse.

**Comment by amaltaro** [comment]
Yes, the distribution of these input blocks is actually something that Ops wanted very much many years ago, which increases resource usage and helps to process large workflows in a shorter period of time. As you say, it becomes a problem when one site has problems and you require that one block to complete a workflow.

IMO, there are many developments that we should make to have a more resilient and performant system, with regards to data input. Just some ideas that came to my mind now:
* block-based input cleanup (not a new idea and definitely something that we must have for Run3)
* mechanism to identify and resolve GQE with stuck input data transfer (or when site is declared offline)
* small input data (needs to be defined!) to have more than 1 copy
* work on workflow priority such that when we start something, we make sure it go through the system as quick as possible, instead of going down in the queue and remaining active for many months in the system.

For this development, I think the best would be to work on making specific input data placement against multiple RSEs with `copies=2`. What do you think?
